src/silx/utils/enum.py

```diff
@@ -30,6 +30,15 @@
 
 import enum
 
+from silx.utils.deprecation import deprecated_warning
+
+deprecated_warning(
+    "Class",
+    "silx.utils.enum.Enum",
+    since_version="2.1.1",
+    replacement="Python built-in Enum class",
+)
+
```

Comment:
This will raise deprecation warnings everywhere silx uses this module.

To keep the compatibility while deprecating it, it needs extra work, something like:
- move the implementation in `_enum.py` and import from `_enum` in this module and at every place it is used in silx
- raise warnings in `from_value`, `members`, `names` and `values` methods

Reply:
Or could I put in in the `__init__` of the class ?